fe1-web/src/features/linked-organizations/reducer/ChallengeReducer.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { Hash } from 'core/objects';
+
+import { ChallengeState } from '../objects/Challenge';
+
+export const CHALLENGE_REDUCER_PATH = 'challenge';
+
+export interface ChallengeReducerState {
+  byLaoId: Record<string, ChallengeState>;
+}
+
+const initialState: ChallengeReducerState = {
+  byLaoId: {},
+};
+
+const challengeSlice = createSlice({
+  name: CHALLENGE_REDUCER_PATH,
+  initialState,
+  reducers: {
+    addChallenge: {
+      prepare(laoId: Hash, challenge: ChallengeState) {
+        return {
+          payload: {
+            laoId: laoId.valueOf(),
+            challenge: challenge,
+          },
+        };
+      },
+      reducer(state, action: PayloadAction<{ laoId: string; challenge: ChallengeState }>) {
+        const { laoId, challenge } = action.payload;
+        state.byLaoId[laoId] = challenge;
```

Comment:
Given this implementation it is more of a `setChallenge` rather than a `addChallenge`, no?